packages/addons-inject/index.js

```diff
@@ -0,0 +1,309 @@
+  // from tests. There might be a way to conditionally enable this, but I had
+  // trouble getting Wrangler vars to work at all.
+  const throwError = headers.get("X-RTD-Throw-Error");
+  if (throwError) {
+    console.log(`Throw error: ${throwError}`);
+  }
+
+  // Get project/version slug from headers inject by El Proxito
+  const projectSlug = headers.get("X-RTD-Project") || "";
+  const versionSlug = headers.get("X-RTD-Version") || "";
+  const resolverFilename = headers.get("X-RTD-Resolver-Filename") || "";
+
+  // Check to decide whether or not inject Addons library. We only do this for
+  // `text/html` content types.
+  if (contentType.includes("text/html")) {
+    // Remove old implementation of our flyout and inject the new addons if the following conditions are met:
+    //
+    // - header `X-RTD-Force-Addons` is present (user opted-in into new beta addons)
+    // - header `X-RTD-Hosting-Integrations` is not present (added automatically when using `build.commands`)
+    //
+    if (forceAddons === "true" && injectHostingIntegrations === "false") {
+      let rewriter = new HTMLRewriter();
+
+      // Remove by selector lookup
+      for (const script of AddonsConstants.removalScripts) {
+        rewriter.on(script, new removeElement());
+      }
+      for (const link of AddonsConstants.removalLinks) {
+        rewriter.on(link, new removeElement());
+      }
+      for (const element of AddonsConstants.removalElements) {
+        rewriter.on(element, new removeElement());
+      }
+
+      // TODO match the pattern above and make this work
+      // NOTE: I wasn't able to reliably remove the "<script>" that parses
+      // the "READTHEDOCS_DATA" defined previously, so we are keeping it for now.
+      //
+      // rewriter.on(readthedocsDataParse, new removeElement())
+      // rewriter.on(readthedocsData, new removeElement())
+
+      return rewriter
+        .on("head", new addPreloads())
+        .on(
+          "head",
+          new addMetaTags(
+            projectSlug,
+            versionSlug,
+            resolverFilename,
+            httpStatus,
+          ),
+        )
+        .on("*", {
+          // This is only used for testing purposes. This is used to mimic an
+          // error being thrown during the request to origin. There are cases
+          // that are difficult to model here, like a large page size timing out
+          // the request.
+          element(element) {
+            if (throwError) {
+              throw new Error("Manually triggered error in transform");
+            }
+          },
+        })
+        .transform(await response.clone());
+    }
+  }
+
+  // Inject Addons if the following conditions are met:
+  //
+  // - header `X-RTD-Hosting-Integrations` is present (added automatically when using `build.commands`)
+  // - header `X-RTD-Force-Addons` is not present (user opted-in into new beta addons)
+  //
+  if (forceAddons === "false" && injectHostingIntegrations === "true") {
```

Comment:
Same as above, only one of these can be true at a time.
```suggestion
  // Inject Addons if the following conditions are met:
  //
  // - header `X-RTD-Hosting-Integrations` is present (added automatically when using `build.commands`)
  // - header `X-RTD-Force-Addons` is not present (user opted-in into new beta addons)
  //
  if (injectHostingIntegrations === "true") {
```

Reply:
I'll leave this be too, I don't want to prune too much logic yet. We can open an issue to track if needed.

Reply:
Also though, I'll note that this logic does feel inaccurate now. I am a bit confused by the variables/headers used here though. Is this saying that when the user has opted out of addons but is using build.commands, they still get addons?

If so, it seems like this logic won't matter. Almost all projects have addons enabled now, save for some that have asked to have addons disabled manually. So almost all projects should skip this logic, and if they set addons to be disabled, it feels weird to magically enable it again.

Shouldn't this be `forceAddons === "true"`?

Reply:
> Is this saying that when the user has opted out of addons but is using build.commands, they still get addons?

I understand you are correct here. Projects using `build.commands` can't disable addons. It was built like that originally to start testing addons on builds we don't have any control.

> Almost all projects have addons enabled now, save for some that have asked to have addons disabled manually. So almost all projects should skip this logic, and if they set addons to be disabled, it feels weird to magically enable it again.

We should probably open an issue to talk more about this and probably remove this code now. I need to think more, tho, before being 100% sure it's like this 😄